src/Tracker/Ajax.php

```
@@ -110,6 +111,17 @@ public static function process()
         \wp_slimstat::set_stat($stat);
 
         if (!empty($data_js['id'])) {
+            // Defense-in-depth: check bot status even for follow-up AJAX events.
+            // The initial pageview (id=empty) goes through Processor::process() which
+            // has the full bot check, but follow-up events skip Processor entirely.
+            // This ensures bots executing JS are still blocked on updates. See #291.
+            if ('on' == \wp_slimstat::$settings['ignore_bots']) {
+                $browser = Browscap::get_browser();
+                if (1 == $browser['browser_type']) {
+                    return Utils::logError(313);
```

PHPStan (2.1.46)

[error] Call to an undefined static method SlimStat\Tracker\Utils::logError().


(staticMethod.notFound)